### ISSUE TITLE
Error: [Reanimated] Mismatch between JavaScript part and native part of Reanimated (3.12.1 vs 3.11.0)

### ISSUE THREAD

**Issue by Gnoot01** [mrousavy/VisionCameraSkiaDemo#3]
os: Windows 11

After cloning the repo, I use "npm install" then "npm start" and encounter this error:
"Error: [Reanimated] Mismatch between JavaScript part and native part of Reanimated (3.12.1 vs 3.11.0)"
and also
"TypeError: Cannot read property 'makeMutable' of undefined"

![a](https://github.com/mrousavy/VisionCameraSkiaDemo/assets/86306864/7992b02c-5b88-443b-807e-24d3eb836998)
![b](https://github.com/mrousavy/VisionCameraSkiaDemo/assets/86306864/b95e4ffd-ca8d-4bcd-95d0-cfe33c710aa5)

**Comment by mantu-bit** [comment]
Same issur

**Comment by wodin** [comment]
I am not 100% sure what would cause your error, however I have an idea:

This repository has a `yarn` lock file instead of an `npm` lock file.

The way reanimated is specified in the `package.json` allows `npm` to install a later version of the JS code if there is no lock file telling it to install a specific version:

https://github.com/mrousavy/VisionCameraSkiaDemo/blob/1d047c16b8266f47b387d3a18ff9f8513eeefc54/package.json#L17

This basically means `npm` is allowed to install 3.11.x or 3.12.x or 3.13.x etc., etc.
But the native code was built against version 3.11.0.

Because there's a `yarn` lock file, if you install the dependencies using `yarn`, you will get the version specified in the lock file:
https://github.com/mrousavy/VisionCameraSkiaDemo/blob/1d047c16b8266f47b387d3a18ff9f8513eeefc54/yarn.lock#L6023-L6024

If you insist on using `npm` instead of `yarn`, you could change the lock file to remove the `^` like this:
```json
    "react-native-reanimated": "3.11.0",
```

tl;dr try using `yarn install` rather than `npm install`
